### PR TITLE
Implement swipe feature between eateries

### PR DESCRIPTION
**Pull request by antoinztte**
## Overview

- Enable users to swipe between multiple eateries when viewing eateries in "Nearest to You" carousel view. Swiping is infinite and users can press the back button anytime to return to the homepage. 

## Changes Made

- Created EateryPageViewController to enable swiping between eateries
- Created new eateryVC for each eatery in Eatery dummyData array, then appended to pages array in EateryPageViewController
- Pushed EateryPageVC in createCarouselView func of HomeModelController upon user press

## Test Coverage

- Manually tested/played around with it
- Used dummyData so that eatery pages show up

## Next Steps (delete if not applicable)

- Plan to apply same changes to stackView
- May change infinite swipe feature depending on feedback from Pod Lead

## Screenshots

Swipe-feature

### CODE REVIEW

File: .DS_Store
Comment (NgaiJustin) [inline .DS_Store:1]: Don't push this file; you should add this to the `.gitignore`. This file contains the custom attributes about the folder that it was in (you probably moved some files around.

File: Eatery Blue/.DS_Store
Comment (NgaiJustin) [inline Eatery Blue/.DS_Store:1]: Same as above.

File: Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    
+}
+
+extension EateryPageViewController: UIPageViewControllerDelegate {
+    
+}
+
+extension EateryPageViewController: UIPageViewControllerDataSource {
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        guard let index = pages.firstIndex(of: viewController), index > 0 else { return pages[pages.count - 1] }
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift:57]: You can just do `let previousIndex = abs((index - 1) % pages.count)` and then `return pages[previousIndex]`

File: Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        
+        return pages[index - 1]
+    }
+    
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerAfter viewController: UIViewController) -> UIViewController? {
+        guard let index = pages.firstIndex(of: viewController), index + 1 < pages.count else { return pages[0] }
+
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift:64]: As above

**Review by NgaiJustin — APPROVED** [review]
Yup, great job! Left a few comments.

**Review by jjennifergu — APPROVED** [review]
Great job on your first PR I'm so proud!! I agree with everything Justin said. Also, just from a design aspect, I don't know if I like the extra bar on the bottom with the swiping dots, maybe ask Kathleen for more advice?

File: Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            eateryVC.setUp(eatery: eatery)
+            pages.append(eateryVC)
+        }
+        
+        setViewControllers([pages[0]], direction: .forward, animated: true, completion: nil)
+    }
+    
+}
+
+extension EateryPageViewController: UIPageViewControllerDelegate {
```
Comment (ReadeP) [inline Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift:51]: If you do not need this delegate you can remove it and remove `delegate = self` in `viewDidLoad`

File: Eatery Blue/Eatery Blue/UI/HomeScreen/HomeModelController.swift [hunk 4]
```diff
@@ -20,7 +20,7 @@ class HomeModelController: HomeViewController {
     private let filterController = EateryFilterViewController()
 
     private var cancellables: Set<AnyCancellable> = []
-
+
```
Comment (ReadeP) [inline Eatery Blue/Eatery Blue/UI/HomeScreen/HomeModelController.swift:23]: remove tabbing

File: Eatery Blue/Eatery Blue/UI/HomeScreen/HomeModelController.swift [hunk 5]
```diff
@@ -139,10 +139,15 @@ class HomeModelController: HomeViewController {
             }
 
             carouselView.addCardView(contentView, buttonPress: { [self] _ in
-                pushViewController(for: eatery)
+                navigationController?.hero.isEnabled = false
+
```
Comment (ReadeP) [inline Eatery Blue/Eatery Blue/UI/HomeScreen/HomeModelController.swift:143]: delete newline

File: Eatery Blue/Eatery Blue/UI/HomeScreen/HomeViewController.swift [hunk 6]
```diff
@@ -92,6 +92,7 @@ class HomeViewController: UIViewController {
         viewController.setUp(eatery: eatery)
         navigationController?.hero.isEnabled = false
         navigationController?.pushViewController(viewController, animated: true)
+
```
Comment (ReadeP) [inline Eatery Blue/Eatery Blue/UI/HomeScreen/HomeViewController.swift:95]: delete newline

**Pull request merged**